configure-ironic.sh

```diff
@@ -64,20 +69,68 @@ mkdir -p /shared/html
 mkdir -p /shared/ironic_prometheus_exporter
 
 HTPASSWD_FILE=/etc/ironic/htpasswd
+# The user can provide HTTP_BASIC_HTPASSWD and HTTP_BASIC_HTPASSWD_RPC. If
+# - we are running conductor and HTTP_BASIC_HTPASSWD_RPC is not set but HTTP_BASIC_HTPASSWD is,
+#   fallback to use HTTP_BASIC_HTPASSWD for RPC, to preserve compatibility.
+# - we are running combined API and conductor, and HTTP_BASIC_HTPASSWD_RPC is set, use it
+# - we are running combined and HTTP_BASIC_HTPASSWD_RPC is unset, but HTTP_BASIC_HTPASSWD is,
+#   i.e. API is authenticated. We want to authenticate RPC by default, but the user might override.
+#   Then try to infere the authentication strategy and credentials from /auth/ironic-rpc/auth-config.
+#   If not present, then generate a username and password, create the config file and set
+#   HTTP_BASIC_HTPASSWD_RPC
 if [ -n "${HTTP_BASIC_HTPASSWD}" ]; then
     printf "%s\n" "${HTTP_BASIC_HTPASSWD}" >"${HTPASSWD_FILE}"
+
+    if [ -z "${HTTP_BASIC_HTPASSWD_RPC:-}" ]; then
+
+        # When running both API and Conductor in the same container, we'll try to get the credentials
```

Comment:
We should just stop doing it, really. It's an artefact of the early days of metal3 that we even allow that.

Reply:
I agree, however, many things in our repo still rely on that script. I have some PR to change it, however, I would still maintain it for a while for compatibility. Some downstream product are relying on it, and might need time to adopt the new changes.